### PR TITLE
update log checkpoint to use debug/checkups interface

### PR DESCRIPTION
**Pull request by zackattack01**
This is a general cleanup and refactor of the log checkpoint code to utilize the shared checkup interface for flare and doctor. The only update required to the interface was to make the `Data` return a `map[string]any` instead of any, to allow us to iterate and format `Data()` more reliably from the log checkpoint routine without adding globally required interface methods at this point.

I am still working on getting tests updated and am breaking that effort into a separate PR to get eyes on this earlier.

<details><summary>Here is a sample of the expected checkpoint logs</summary>
<p>

```
{
  "architecture": "arm64",
  "caller": "rungroup.go:55",
  "checkup": "Platform",
  "component": "log checkpoint",
  "platform": "darwin",
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:22.19229Z"
}
{
  "caller": "rungroup.go:55",
  "checkup": "Launcher Version",
  "component": "log checkpoint",
  "launcher_version": "1.1.2-9-gfe787de-dirty",
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:22.195288Z",
  "tufServer": "https://example.com/",
  "update_channel": "nightly"
}
{
  "bbolt_db_size": "3817472",
  "caller": "rungroup.go:55",
  "checkup": "Host Info",
  "component": "log checkpoint",
  "hostname": "host.localdomain",
  "keyinfo": "map[local_key:Weeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee==]",
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:22.198365Z",
  "user_desktop_processes": "map[501:/usr/bin/sudo [socket: /var/kolide-k2/k2device-preprod.kolide.com/desktop_501/desktop.sock_6743, started: 2023-10-06 19:03:23.210173 +0000 UTC, last_health_check: 2023-10-06 19:05:17.208848 +0000 UTC])]"
}
{
  "caller": "rungroup.go:55",
  "checkup": "Check communication with Kolide",
  "component": "log checkpoint",
  "control": "beeeeeeeeeeeeeeeeeeeeeeeeeeeeeee\n",
  "device": "booooooooooooooooooooooooooooop\n",
  "log": "expected status 200, got 404", // ? let me know if anyone knows the expected response from the log endpoint here
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:27.392102Z"
}
{
  "apple.com": "2620:149:af0::10,17.253.144.10",
  "caller": "rungroup.go:55",
  "checkup": "DNS Resolution",
  "component": "log checkpoint",
  "google.com": "2607:f8b0:4006:81d::200e,142.250.80.110",
  "k2control-preprod.kolide.com": "54.161.241.46,52.202.168.65,18.205.222.128,54.237.133.81",
  "k2device-preprod.kolide.com": "54.161.241.46,18.205.222.128,52.202.168.65,54.237.133.81",
  "lookup_attempts": 5,
  "lookup_successes": 5,
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:27.466498Z",
  "tuf.kolide.com": "34.149.84.181"
}
{
  "caller": "rungroup.go:55",
  "checkup": "Server Data",
  "component": "log checkpoint",
  "device_id": "111111",
  "munemo": "beepboopbay",
  "organization_id": "100",
  "remote_ip": "9.99.99.9",
  "session_pid": 36331,
  "ts": "2023-10-06T19:05:27.466752Z"
}

```
</p>
</details> 

<details><summary>And here is the expected doctor output</summary>
<p>

```
zack:launcher❱ sudo launcher doctor
        Platform: platform: darwin, architecture: arm64
        Launcher Version: launcher_version 1.1.2-11-gca7d851-dirty
        Host Info: hostname: host.localdomain
✅      Process Report: found 5 kolide processes
✅      Root directory contents: root directory (/var/kolide-k2/k2device-preprod.kolide.com) contains 26 files
✅      Check communication with Kolide: successfully connected to device and control server
✅      Logs: debug.json is 967355 bytes, and was last modified at 2023-10-06 16:20:01.590527916 -0400 EDT
✅      Binary directory contents: binary directory (/usr/local/kolide-k2) contains 2 files
✅      Launchd: state is running
✅      Enrollment Secret: claim for beeboopbay
✅      Network Report: launcher can listen on local network
✅      Osquery: osqueryd version 5.8.1
✅      Launcher Flags: launcher.flags exists and is parsable
✅      Quarantine: no quarantine directories found
✅      System Time: system time is within 5.000000 minutes of server date header, delta = 0.028745 minutes
✅      DNS Resolution: successfully resolved 5/5 hosts
✅      Notary Version: Successfully gathered notary version 3726 from https://example.com/releases.json
✅      Tuff Version: Successfully gathered release version launcher/darwin/universal/launcher-1.1.2-3-g29bc19e.tar.gz from https://example.com/repository/targets.json
✅      Notable Files: No notable directories were detected
⚠️       Server Data: no bbolt DB connection in knapsack
✅      Server Version: https://example.com/version returned version response: 200 OK 3a3fb4f9843f83c2fb0893a6c87cad7b8f139295

✅      Osquery Data: hardware_model: Mac14,10, hardware_serial: 99999999, hardware_uuid: 999999-9999-999999-999999, hardware_vendor: Apple Inc., os_build: 23A344, os_name: macOS, os_version: 14.0

Checkups with warnings:
        * Server Data
```

</p>
</details> 

This looks like a lot more changes than there are, a lot of the effort was merging/deduplicating code while moving things over so many of the changes are clean deletions here

### CODE REVIEW

File: pkg/debug/checkups/checkpoint.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		logValues := []interface{}{"checkup", checkup.Name()}
+		for k, v := range checkup.Data() {
+			logValues = append(logValues, k, c.summarizeData(v))
+		}
+
+		c.logger.Log(logValues...)
+	}
+}
+
+func (c *checkPointer) summarizeData(data any) any {
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/checkpoint.go:74]: Nice

File: pkg/debug/checkups/files.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+type filesCheckup struct {
+	k       types.Knapsack
+	status  Status
+	summary string
+	data    map[string]any
+}
+
+func (fc *filesCheckup) Data() map[string]any  { return fc.data }
+func (fc *filesCheckup) ExtraFileName() string { return "" }
+func (fc *filesCheckup) Name() string          { return "Notable Files" }
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/files.go:24]: I think the name is preexisting, but since this check exists to determine whether there could be a secondary/separate osquery installation that interferes with the Kolide/launcher installation, maybe a good time to rename this to "Non-launcher osquery installation" or "Unexpected osquery config files" or something else more specific?

File: pkg/debug/checkups/notary.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+	notaryEndpoint := fmt.Sprintf("%s/v2/kolide/launcher/_trust/tuf/targets/releases.json", nc.k.NotaryServerURL())
+	notaryUrl, err := parseUrl(nc.k, notaryEndpoint)
+	if err != nil {
+		return err
+	}
+
+	response, err := fetchNotaryVersion(httpClient, notaryUrl)
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/notary.go:49]: It might be interesting to compare the metadata version for releases on notary.kolide.co with the local tuf repo on disk (same file can be found at `/var/kolide-k2/k2device-preprod.kolide.com/launcher-tuf/targets/releases.json` for darwin/linux -- same dir should be should be in `data` on Windows iirc). That would tell us "does this installation of launcher has the most recent version of notary data available". But maybe more effort than it's worth for the legacy autoupdater.

File: pkg/debug/checkups/osq_data.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+// osqueryInteractive execs osquery and parses the output to gather some basic host info.
+// it was done this way to avoid bringing Querier into knapsack for a task that will only be run
+// during flare or doctor
+func (odc *osqDataCollector) osqueryInteractive(ctx context.Context, query string) (map[string]string, error) {
+	cmdCtx, cmdCancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cmdCancel()
+
+	osqPath := odc.k.LatestOsquerydPath(ctx)
+	cmd := exec.CommandContext(cmdCtx, osqPath, "-S", "--json")
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/osq_data.go:77]: Would we want to use `runsimple` (as in https://github.com/kolide/launcher/pull/1379/files) instead?

File: pkg/debug/checkups/osquery.go [hunk 5]
```diff
@@ -61,26 +60,19 @@ func (o *osqueryCheckup) version(ctx context.Context) (string, error) {
...
-	case "windows":
-		launcherPath = `C:\Program Files\Kolide\Launcher-kolide-k2\bin\launcher.exe`
-	}
-
 	cmdCtx, cmdCancel := context.WithTimeout(ctx, 20*time.Second)
 	defer cmdCancel()
 
-	cmd := exec.CommandContext(cmdCtx, launcherPath, "interactive")
+	osqPath := o.k.LatestOsquerydPath(ctx)
+	cmd := exec.CommandContext(cmdCtx, osqPath, "-S")
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/osquery.go:67]: The point of running `launcher interactive` instead of osquery directly was to test (and time) the launcher <=> osquery communication piece -- I'm not sure we want to change this, given that?

File: pkg/debug/checkups/server_data.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/kolide/launcher/pkg/agent/types"
+	"go.etcd.io/bbolt"
+)
+
+var serverProvidedDataKeys = []string{
+	"munemo",
+	"organization_id",
+	"device_id",
+	"remote_ip",
+	"tombstone_id",
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/server_data.go:19]: Ooh interesting, is this new?

**Review by RebeccaMahany — COMMENTED** [review]
Made it partway through, will finish reviewing next week -- looks great so far, all the new data and organization is 🔥

File: pkg/debug/checkups/files.go (hunk 2 above)
Comment (zackattack01) [reply pkg/debug/checkups/files.go:24]: that makes sense to me! maybe `Osquery Conflicts`? just trying to think of something shortish since these turn into the dir names

File: pkg/debug/checkups/osq_data.go (hunk 4 above)
Comment (zackattack01) [reply pkg/debug/checkups/osq_data.go:77]: ooh yeah that does look cleaner, i'll check that out thank you!

File: pkg/debug/checkups/server_version.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+func (svc *serverVersionCheckup) Summary() string       { return svc.summary }
+
+func (svc *serverVersionCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	svc.data = make(map[string]any)
+	if !svc.k.KolideHosted() {
+		return nil
+	}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+	serverEndpoint := fmt.Sprintf("%s/version", svc.k.KolideServerURL())
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/server_version.go:35]: Is this for helping us debug k2 <=> launcher issues, so we know what version of k2 was running when we saw an issue, or for something else?

File: pkg/debug/checkups/tuf.go [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	tc.data = make(map[string]any)
+	if !tc.k.Autoupdate() {
+		return nil
+	}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+
+	tuffEndpoint := fmt.Sprintf("%s/repository/targets.json", tc.k.TufServerURL())
```
Comment (RebeccaMahany) [inline pkg/debug/checkups/tuf.go:51]: `tufEndpoint`

Comment (RebeccaMahany) [reply pkg/debug/checkups/tuf.go:51]: Also below, `tufUrl`, `fetchTufVersion`

File: pkg/debug/checkups/notary.go (hunk 3 above)
Comment (RebeccaMahany) [reply pkg/debug/checkups/notary.go:49]: Thinking on it more, I think this falls into "interesting but not worth doing" haha

File: pkg/debug/checkups/server_version.go (hunk 7 above)
Comment (zackattack01) [reply pkg/debug/checkups/server_version.go:35]: yes i believe this is the original intent because we separately have resolution and connectivity checks for this host- but auditing this before answering I'm realizing this info will already be included in the connectivity check (which looks at `/version` by default), so I think we can pull this out cleanly. good question

File: pkg/debug/checkups/tuf.go (hunk 8 above)
Comment (zackattack01) [reply pkg/debug/checkups/tuf.go:51]: oh wow I will clean that all up, thank you!

File: pkg/debug/checkups/notary.go [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+		nc.data[notaryUrl.String()] = err.Error()
+		nc.status = Failing
+		nc.summary = fmt.Sprintf("Unable to gather notary version response from %s", notaryUrl.String())
+	} else {
+		nc.data[notaryUrl.String()] = response
+		nc.status = Passing
+		nc.summary = fmt.Sprintf("Successfully gathered notary version %s from %s", response, notaryUrl.String())
+	}
+
+	return nil
```
Comment (James-Pickett) [inline pkg/debug/checkups/notary.go:60]: reduce a little nesting

```suggestion
	if err != nil {
		nc.data[notaryUrl.String()] = err.Error()
		nc.status = Failing
		nc.summary = fmt.Sprintf("Unable to gather notary version response from %s", notaryUrl.String())
		return nil
	}
	
        nc.data[notaryUrl.String()] = response
        nc.status = Passing
        nc.summary = fmt.Sprintf("Successfully gathered notary version %s from %s", response, notaryUrl.String())
	return nil
```

**Review by James-Pickett — COMMENTED** [review]
Nice work!

File: pkg/debug/checkups/host.go [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+	} else {
+		keyinfo["local_key"] = fmt.Sprintf("error marshalling local key (startup is sometimes weird): %s", err)
+	}
+
+	// We don't always have hardware keys. Move on if we don't
+	if agent.HardwareKeys().Public() == nil {
+		return keyinfo
+	}
+
+	if hardwareKeyDer, err := x509.MarshalPKIXPublicKey(agent.HardwareKeys().Public()); err == nil {
```
[the remark this answers is not in the document]
Comment (zackattack01) [reply pkg/debug/checkups/host.go:95]: oh good catch, I will update this to work more like the check above it. thank you!

File: pkg/debug/checkups/notary.go (hunk 9 above)
Comment (zackattack01) [reply pkg/debug/checkups/notary.go:60]: nice, love it! I will add this in after the other updates thank you

File: pkg/debug/checkups/osquery.go (hunk 5 above)
Comment (zackattack01) [reply pkg/debug/checkups/osquery.go:67]: when I had discussed taking this idea to use for the osq_data check @directionless had pointed out it would be preferable to invoke osq interactive directly (vs having launcher flare invoke launcher interactive and so on). I had agreed this seemed more direct but likely overstepped updating this existing check as well- any preference here @directionless?

Comment (zackattack01) [reply pkg/debug/checkups/osquery.go:67]: discussed offline, moving this back to the original `launcher interactive` call to preserve the original intent

**Review by James-Pickett — APPROVED** [review]

**Review by RebeccaMahany — APPROVED** [review]

File: pkg/debug/checkups/checkups.go [hunk 11]
```diff
@@ -71,7 +71,7 @@ type checkupInt interface {
 	ExtraFileName() string                                // If this checkup will have extra data, what name should it use in flare
 	Summary() string                                      // Short summary string about the status
 	Status() Status                                       // State of this checkup
-	Data() any                                            // What data objects exist, if any
+	Data() map[string]any                                 // What data objects exist, if any
```
Comment (directionless) [inline pkg/debug/checkups/checkups.go:74]: Why the change? Though I guess it's fine

File: pkg/debug/checkups/checkpoint.go [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+// logger is an interface that allows mocking of logger
+type (
+	logger interface {
+		Log(keyvals ...interface{}) error
+	}
+
+	checkPointer struct {
```
Comment (directionless) [inline pkg/debug/checkups/checkpoint.go:21]: As an FYI I think of this as "Log Checkpoint" Not "Check Point"

File: pkg/debug/checkups/checkpoint.go [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+	c.interrupt <- struct{}{}
+}
+
+func (c *checkPointer) Once(ctx context.Context) {
+	checkups := checkupsFor(c.knapsack, logSupported)
+
+	for _, checkup := range checkups {
+		checkup.Run(ctx, io.Discard)
+
+		logValues := []interface{}{"checkup", checkup.Name()}
```
Comment (directionless) [inline pkg/debug/checkups/checkpoint.go:65]:
```suggestion
		logValues := []interface{}{"checkup", checkup.Name(), "summary" checkup.Summary(), "status", checkup,Status()}
```

I think?

File: pkg/debug/checkups/checkpoint.go [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+		for k, v := range checkup.Data() {
+			logValues = append(logValues, k, c.summarizeData(v))
+		}
+
+		c.logger.Log(logValues...)
```
Comment (directionless) [inline pkg/debug/checkups/checkpoint.go:70]: I thought logger would serialize to json for us. Can you do:
```go
c.logger.Log(
  "checkup", checkup.Name(),
  "summary" checkup.Summary(),
  "status", checkup,Status(),
  "data", c.Data(),
  )
  ```

File: pkg/debug/checkups/dns.go [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+	dc.data["lookup_successes"] = successCount
+
+	dc.summary = fmt.Sprintf("successfully resolved %d/%d hosts", successCount, attemptedCount)
+
+	return nil
+}
+
+func resolveHost(resolver *net.Resolver, host string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
```
Comment (directionless) [inline pkg/debug/checkups/dns.go:81]: You should pass in ctx instead of making a new one.

File: pkg/debug/checkups/dns.go [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(ips) == 0 {
+		return "", fmt.Errorf("host was valid but did not resolve")
+	}
+
+	return strings.Join(ips, ","), nil
```
Comment (directionless) [inline pkg/debug/checkups/dns.go:93]: Can we leave this as []string

File: pkg/debug/checkups/host.go [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+func (hc *hostInfoCheckup) ExtraFileName() string { return "" }
+func (hc *hostInfoCheckup) Name() string          { return "Host Info" }
+func (hc *hostInfoCheckup) Status() Status        { return hc.status }
+func (hc *hostInfoCheckup) Summary() string       { return hc.summary }
+
+func (hc *hostInfoCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	hc.data = make(map[string]any)
+	hc.data["hostname"] = hostName()
+	hc.data["keyinfo"] = agentKeyInfo()
+	hc.data["bbolt_db_size"] = hc.bboltDbSize()
```
Comment (directionless) [inline pkg/debug/checkups/host.go:37]: We have a `bboltdbCheckup`. This should be there, not here.

File: pkg/debug/checkups/host.go [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+	desktopProcesses := runner.InstanceDesktopProcessRecords()
+	hc.data["user_desktop_processes"] = desktopProcesses
```
Comment (directionless) [inline pkg/debug/checkups/host.go:39]: Desktop feels like it should be it's own checkup

File: pkg/debug/checkups/host.go [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+
+	if runtime.GOOS == "windows" {
+		hc.data["in_modern_standby"] = hc.k.InModernStandby()
+	}
```
Comment (directionless) [inline pkg/debug/checkups/host.go:43]: I think this belongs in `powerCheckup`

File: pkg/debug/checkups/osq_config_conflicts.go [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+			for i, file := range files {
+				fileToLog[i] = file.Name()
+			}
+
+			occ.data[dirname] = fmt.Sprintf("contains: %s", strings.Join(fileToLog, ", "))
+		}
+	}
+
+	if dirNotEmpty {
+		occ.status = Failing
```
Comment (directionless) [inline pkg/debug/checkups/osq_config_conflicts.go:68]: This should be a Warning

File: pkg/debug/checkups/osq_config_conflicts.go [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+		occ.summary = "At least one notable directory is present and non-empty"
+	} else if dirHasError {
+		occ.status = Erroring
+		occ.summary = "At least one notable directory is present and could not be read"
+	} else if dirExists {
+		occ.status = Warning
```
Comment (directionless) [inline pkg/debug/checkups/osq_config_conflicts.go:74]: This should probably be Informational

File: pkg/debug/checkups/osq_data.go [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+	odc.status = Passing
+	odc.summary = strings.Join(data, ", ")
+
+	return nil
+}
+
+// osqueryInteractive execs osquery and parses the output to gather some basic host info.
+// it was done this way to avoid bringing Querier into knapsack for a task that will only be run
+// during flare or doctor
+func (odc *osqDataCollector) osqueryInteractive(ctx context.Context, query string) (map[string]string, error) {
```
Comment (directionless) [inline pkg/debug/checkups/osq_data.go:73]: This is not interactive.

File: pkg/debug/checkups/server_data.go [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+func (sdc *serverDataCheckup) Data() map[string]any  { return sdc.data }
+func (sdc *serverDataCheckup) ExtraFileName() string { return "" }
+func (sdc *serverDataCheckup) Name() string          { return "Server Data" }
+func (sdc *serverDataCheckup) Status() Status        { return sdc.status }
+func (sdc *serverDataCheckup) Summary() string       { return sdc.summary }
+
+func (sdc *serverDataCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	db := sdc.k.BboltDB()
```
Comment (directionless) [inline pkg/debug/checkups/server_data.go:36]: This should not use BboltDB. (basically, almost nothing should)

You can get the KV store with `k.ServerProvidedDataStore()` and the `store.Get([]byte("device_id"))` or whatever

Comment (directionless) [reply pkg/debug/checkups/server_data.go:36]: We're trying to remove all bbolt from the codebase, and push it through kv interfaces

File: pkg/debug/checkups/server_data.go [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+
+		return nil
+	}); err != nil {
+		sdc.status = Erroring
+		sdc.data["error"] = err.Error()
+		sdc.summarize()
+		return nil
+	}
+
+	sdc.status = Passing
```
Comment (directionless) [inline pkg/debug/checkups/server_data.go:68]: <thinks>

I think if we have a device_id, than we're passing, and if we don't, we're failing?

File: pkg/debug/checkups/server_data.go [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+
+	return nil
+}
+
+func (sdc *serverDataCheckup) summarize() {
+	summary := make([]string, 0)
+
+	for k, v := range sdc.data {
+		summary = append(summary, fmt.Sprintf("%s: %s", k, v))
+	}
```
Comment (directionless) [inline pkg/debug/checkups/server_data.go:79]: I don't think I'd expose these in the summary. 

Summary is meant for `doctor` which is eventually designed to be end user facing. I'd probably print something about having server data or not.

File: pkg/debug/checkups/tuf.go [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+
+	tufEndpoint := fmt.Sprintf("%s/repository/targets.json", tc.k.TufServerURL())
+	tufUrl, err := parseUrl(tc.k, tufEndpoint)
+	if err != nil {
+		return err
+	}
+
+	response, err := tc.fetchTufVersion(httpClient, tufUrl)
```
Comment (directionless) [inline pkg/debug/checkups/tuf.go:57]: I've seen this in a couple of spots -- havingt a helper for the fetch makes sense, but why generate the URL up here, and not in the same helper?

**Review by directionless — COMMENTED** [review]
Nice. I know this merged, but I've got comments :)
